domain/application/state/application.go

```diff
@@ -205,26 +205,6 @@ WHERE name = $applicationName.name
 	return coreapplication.ID(appID.ID), nil
 }
 
-// GetApplicationID returns the ID for the named application, returning an error
-// satisfying [applicationerrors.ApplicationNotFound] if the application is not found.
-func (st *ApplicationState) GetApplicationID(ctx context.Context, name string) (coreapplication.ID, error) {
```

Comment:
Remove the offending state method. This was calling a transaction in a transaction. This is clearly undefined behaviour.

Reply:
Ah good catch, I had meant to use an atomic context here but missed it as I was converting the original POC across.